nautobot/dcim/factory.py

```diff
@@ -99,6 +99,156 @@ def get_rack_reservation_units(obj):
     return [unit for unit in available_units if unit not in unavailable_units][:1]
 
 
+class PlatformFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Platform
+        exclude = ("has_manufacturer", "has_description", "has_napalm_args")
+
+    # This dictates `name` and `napalm_driver`.
+    has_manufacturer = NautobotBoolIterator()
+
+    name = factory.Maybe(
+        "has_manufacturer",
+        factory.LazyAttributeSequence(lambda o, n: f"{o.manufacturer.name} Platform {n + 1}"),
+        factory.Sequence(lambda n: f"Fake Platform {n}"),
+    )
+
+    manufacturer = factory.Maybe("has_manufacturer", random_instance(Manufacturer), None)
+
+    # If it has a manufacturer, it *might* have a napalm_driver.
+    napalm_driver = factory.Maybe(
+        "has_manufacturer",
+        factory.LazyAttribute(lambda o: random.choice(NAPALM_DRIVERS.get(o.manufacturer.name, [""]))),
+        "",
+    )
+
+    has_napalm_args = NautobotBoolIterator()
+    napalm_args = factory.Maybe(
+        "has_napalm_args", factory.Faker("pydict", nb_elements=2, value_types=[str, bool, int]), None
+    )
+
+    has_description = NautobotBoolIterator()
+    network_driver = factory.Maybe(
+        "has_manufacturer",
+        factory.LazyAttribute(lambda o: random.choice(NETWORK_DRIVERS.get(o.manufacturer.name, [""]))),
+        "",
+    )
+
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+
+class DeviceFactory(PrimaryModelFactory):
+    class Meta:
+        model = Device
+        exclude = (
+            "has_tenant",
+            "has_platform",
+            "has_serial",
+            "has_asset_tag",
+            "has_device_redundancy_group",
+            "has_comments",
+        )
+
+    device_type = random_instance(DeviceType, allow_null=False)
+    status = random_instance(
+        lambda: Status.objects.get_for_model(Device),
+        allow_null=False,
+    )
+    role = random_instance(
+        lambda: Role.objects.get_for_model(Device),
+        allow_null=False,
+    )
+    location = random_instance(
+        lambda: Location.objects.filter(location_type__content_types=ContentType.objects.get_for_model(Device)),
+        allow_null=False,
+    )
+    name = factory.LazyAttributeSequence(lambda o, n: f"{o.device_type.model} Device - {o.location} - {n + 1}")
+
+    has_tenant = NautobotBoolIterator()
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+    has_platform = NautobotBoolIterator()
+    platform = factory.Maybe(
+        "has_platform",
+        factory.SubFactory(
+            PlatformFactory,
+            manufacturer=factory.LazyAttribute(lambda o: o.factory_parent.device_type.manufacturer),
+        ),
+    )
```

Comment:
This makes it so we create a new Platform for each Device that gets created, doesn't it? Would be better to use one of the existing Platforms instead, I think?